package/MDAnalysis/analysis/base.py

```diff
@@ -310,15 +252,119 @@ def _conclude(self):
     .. versionchanged:: 2.0.0
         Added :attr:`results`
 
+    .. versionadded:: 2.8.0
+        Added ability to run analysis in parallel using either a
+        built-in backend (`multiprocessing` or `dask`) or a custom
+        `backends.BackendBase` instance with an implemented `apply` method
+        that is used to run the computations.
     """
 
+    @classmethod
+    def get_supported_backends(cls):
+        """Tuple with backends supported by the core library for a given class.
+        User can pass either one of these values as ``backend=...`` to
+        :meth:`run()` method, or a custom object that has ``apply`` method
+        (see documentation for :meth:`run()`):
+
+         - 'serial': no parallelization
+         - 'multiprocessing': parallelization using `multiprocessing.Pool`
+         - 'dask': parallelization using `dask.delayed.compute()`. Requires
+           installation of `mdanalysis[dask]`
+
+        If you want to add your own backend to an existing class, pass a
+        :class:`backends.BackendBase` subclass (see its documentation to learn
+        how to implement it properly), and specify ``unsupported_backend=True``.
+
+        Returns
+        -------
+        tuple
+            names of built-in backends that can be used in :meth:`run(backend=...)`
+        """
+        return ("serial",)
+
+    @classmethod
+    def _is_parallelizable(cls):
```

Comment:
Why is this a method? Can't we make it a managed attribute ... or just a class attribute?

I find it confusing when `is_XXX` has a call signature.

Furthermore, in `transformations.TransformationBase` we have the attribute `parallelizable` https://github.com/MDAnalysis/mdanalysis/blob/45fb664a4e053772c3d5cf541b2705e270a7f2a6/package/MDAnalysis/transformations/base.py#L108 — I would prefer having a common API, i.e., turn `_is_parallelizable` into the attribute `parallelizable`. It should _not_ be a class attribute of the base class to avoid someone accidentally changing it in a derived class without having overriden it first (which would ultimately mess up the parent class) — or make it a managed attribute.

Reply:
I see, it's a valid concern.

It first was a `@property @classmethod`, but:

>Class properties are deprecated in Python 3.11 and will not be supported in Python 3.13 // Pylance

So I changed it to be a `@classmethod` only. The (initial) reason why it's not an attribute is that there's no such thing as `@classattribute`, and I wanted `parallelizable` to be a readonly property of a class, and not of its instance, since we don't always know the parameters of the class `__init__` to check if it's parallelizable.

As for now, I see two possible solutions to it (see also a [gist](https://gist.github.com/marinegor/7edd5787aa03b1896855c5ac3422cffb)):
 - have `__is_parallelizable` attribute of all `AnalysisBase` subclasses, and a `@property` that returns it. The downside is that someone could change it to be `True`, and everything will break -- but I guess that's what you get when you change double-underscore parameter which isn't supposed to be changed.
 - rename `_is_parallelizable` to something like `get_parallelizability()` and leave it as `@classmethod`, hence maintaining the semantics that you prefer, i.e. a verb-containing method is callable.
 
Also, we wanted to have a default `AnalysisBase` attribute/property/method/whatever to make parallelization opt-in, so I can't see a way to make it instantiated in `__init__`.

Reply:
Let's just not mess around with double-underscore attributes. If we _really_ wanted to we could un-do the dunder-name mangling but given that Python just does not allow us to really lock away anything, just use convention and if users want to mess up things then they are allowed to. I suggest we use properties with simple class attributes in the following way:
```python
class AnalysisBase:
    # class authors: override _analysis_algorithm_is_parallelizable 
    # in derived classes and only set to True if you have confirmed 
    # that your algorithm works reliably when parallelized with 
    # the split-apply-combine approach (see docs)
    _analysis_algorithm_is_parallelizable = False

    def __init__(self, traj, **kwargs):
        ...

    @property
    def parallelizable(self):
        """Read-only attribute that indicates if this analysis can be
        performed in parallel.

        .. warning:: 
           The author of the code has to determine if the algorithm
           can be run in parallel. If users change this setting in any 
           way then **wrong results** will likely be produced.
        """
        return self._analysis_algorithm_is_parallelizable
```
and then a parallelizable derived class can just use
```python
class MyParallelAnalysis(AnalysisBase):
        _analysis_algorithm_is_parallelizable = True
```

Enterprising users will be able to change `_analysis_algorithm_is_parallelizable` but we tell them they shouldn't so that becomes their problem, not ours.

Using an un-documented single underscore attr with a long name should also make clear that this is not something that users ought to touch.

At the end of the day, all of Python APIs are just conventions so I am ok with doing this by convention.